### PR TITLE
  Unmanaged resource leak

### PR DESCRIPTION
**Pull request by Brain2000**

### CODE REVIEW

**Comment by EamonNerbonne** [comment]
@bp74 is this mergable?

**Comment by Brain2000** [comment]
I hope so. I merged all the changes into my fork, added this, then made the pull request. It's only my second pull requests, so perhaps I did it incorrectly or there is a better way to do it?

File: Zstandard.Net/ZstandardStream.cs [hunk 1]
```diff
@@ -156,6 +156,7 @@ protected override void Dispose(bool disposing)
 
             if (this.isDisposed == false)
             {
+                if (!this.isClosed) ReleaseResources(flushStream: false);
```
Comment (EamonNerbonne) [inline Zstandard.Net/ZstandardStream.cs:159]: If I understand correctly, this means that if a compressing `ZstandardStream` is used in a conventional `using` block, then after the block completes and implicitly `Dispose` has been called - that the data written into the `ZstandardStream` instance is not necessarily written to the underlying stream?

**Comment by EamonNerbonne** [comment]
I took a peak at what GZipStream does; and that is implemented by DeflateStream, and *that* calls the following code on dispose: https://github.com/dotnet/corefx/blob/b8bc4ff80c5f7baa681e8a569d367356957ba78a/src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs#L585

In other words, if it's being finalized, there's no flushing, but if it's being disposed, there *is* flushing.

I think I'd consider that to be the expected behavior, if by nothing other than example.

File: Zstandard.Net/ZstandardStream.cs (hunk 1 above)
Comment (Brain2000) [reply Zstandard.Net/ZstandardStream.cs:159]: This is the same behavior prior to my change. What I added was the ReleaseResources( ), which at least frees the memory stopping the leak.

**Pull request merged**